rust/src/jit_proxy_client.rs

```diff
@@ -0,0 +1,164 @@
+            let tx_builder = self.drift_client.init_tx(&self.drift_client.wallet().default_sub_account(), false).await.unwrap();
+            let program_data = tx_builder.program_data();
+            let account_data = tx_builder.account_data();
+            
+            let wallet =  self.drift_client.wallet();
+
+            let writable_markets = match order.market_type {
+                MarketType::Perp => {
+                    vec![MarketId::perp(order.market_index)]
+                }
+                MarketType::Spot => {
+                    vec![MarketId::spot(order.market_index), MarketId::QUOTE_SPOT]
+                }
+            };
+
+            let mut accounts = build_accounts(
+                program_data,
+                jit_proxy::accounts::Jit {
+                    state: *state_account(),
+                    user: wallet.default_sub_account(),
+                    user_stats: drift_sdk::Wallet::derive_stats_account(wallet.authority(), &PROGRAM_ID),
+                    taker: params.taker_key,
+                    taker_stats: params.taker_stats_key,
+                    authority: *wallet.authority(),
+                    drift_program: PROGRAM_ID,
+                },
+                &[&params.taker, account_data],
+                &[],
+                &writable_markets.as_slice(),
+            );
+
+            if let Some(referrer_info) = params.referrer_info {
+                accounts.push(AccountMeta::new(referrer_info.referrer(), false));
+                accounts.push(AccountMeta::new(referrer_info.referrer_stats(), false));
+            }
+
+            if order.market_type == MarketType::Spot {
+                let spot_market_vault = self.drift_client.get_spot_market_info(order.market_index).await?.vault;
+                let quote_spot_market_vault = self.drift_client.get_spot_market_info(QUOTE_SPOT_MARKET_INDEX).await?.vault;
+                accounts.push(AccountMeta::new_readonly(spot_market_vault, false));
+                accounts.push(AccountMeta::new_readonly(quote_spot_market_vault, false));
+            }      
+
+            let jit_params = jit_proxy::instructions::JitParams {
+                taker_order_id: params.taker_order_id,
+                max_position: params.max_position,
+                min_position: params.min_position,
+                bid: params.bid,
+                ask: params.ask,
+                price_type: params.price_type.unwrap(),
+                post_only: params.post_only,
+            };     
+
+            let ix = Instruction {
+                program_id: jit_proxy::id(),
+                accounts,
+                data: jit_proxy::instruction::Jit {
+                    params: jit_params,
+                }.data(),
+            };
+
+            let lut = program_data.lookup_table.clone();
+
+            let message = v0::Message::try_compile(
+                &self.drift_client.wallet().authority(),
+                &[ix],
```

Comment:
should make this a builder pattern or let user pass in compute budget params, you're probably gonna run out of CUs if you're actually jit filling 8 markets and filling up your positions